solr/solrj/src/java/org/apache/solr/client/solrj/impl/HttpClientBuilderFactory.java

```diff
@@ -36,7 +35,7 @@
    *                by configured (optional).
    * @return the {@linkplain SolrHttpClientBuilder}
    */
-  public SolrHttpClientBuilder getHttpClientBuilder(Optional<SolrHttpClientBuilder> builder);
+  public SolrHttpClientBuilder getHttpClientBuilder(SolrHttpClientBuilder builder);
```

Comment:
I'm wondering if this is a change we want to do in 9.0 only?

Reply:
Yea, since we're changing APIs here, it would have to probably be 9 only